### PR TITLE
consistency fixes and other static analysis review

### PR DESCRIPTION
**Pull request by mjrerle**

### CODE REVIEW

File: src/main/java/p2/dao/IInterestDAO.java [hunk 1]
```diff
@@ -3,22 +3,10 @@
...
-
-	public int insert(Interest interest);
-
-	public boolean update(Interest interest);
-
-	public List<Interest> findAll();
-
-	public Interest findById(int id);
-
-	public boolean deleteById(int id);
```
Comment (vajirakarunathilake) [inline src/main/java/p2/dao/IInterestDAO.java:18]: Why You remove these methods?

File: src/main/java/p2/dao/IInterestDAO.java [hunk 2]
```diff
@@ -3,22 +3,10 @@
...
 	
 	public List<Interest> findByProductId(int productId);
-
-	public int changeStatus(int productId, int status);
```
Comment (vajirakarunathilake) [inline src/main/java/p2/dao/IInterestDAO.java:22]: Why you remove this method?

File: src/main/java/p2/dao/impl/FavoriteDAO.java [hunk 3]
```diff
@@ -1,30 +1,32 @@
...
 public class FavoriteDAO extends GenericDAO<Favorite> implements IFavoriteDAO {
 	
 	@SuppressWarnings("unchecked")
 	public List<Favorite> findByUser(User user){
 		Session session = HibernateUtil.getSession();
-		List<Favorite> f = null;
+		List<Favorite> favorites = null;
 
 		try {
-			f = session.createQuery("FROM favorites WHERE user.id = " + user.getId()).list();
```
Comment (vajirakarunathilake) [inline src/main/java/p2/dao/impl/FavoriteDAO.java]: This is working with Hibernate.

File: src/main/java/p2/dao/impl/InterestDAO.java [hunk 4]
```diff
@@ -20,8 +19,8 @@
 		List<Interest> list = null;
 
 		try {
-			Query query = session.createQuery("FROM Interest WHERE product.id = :productId");
```
Comment (vajirakarunathilake) [inline src/main/java/p2/dao/impl/InterestDAO.java:23]: This is working. What is the problem?

File: src/main/java/p2/dao/impl/UserDAO.java [hunk 5]
```diff
@@ -1,25 +1,29 @@
...
 
 public class UserDAO extends GenericDAO<User> implements IUserDAO {
+	private static Logger logger = Glogger.logger;
 
 	public User findByEmailAndPassword(String email, String password) {
 		Session session = HibernateUtil.getSession();
 		User user = null;
 
 		try {
-			Query query = session.createQuery("FROM Interest WHERE email = :email" + "AND password = :password");
```
Comment (vajirakarunathilake) [inline src/main/java/p2/dao/impl/UserDAO.java:18]: This is work perfect. This is writing pattern

File: src/main/java/p2/dao/IInterestDAO.java (hunk 1 above)
Comment (mjrerle) [reply src/main/java/p2/dao/IInterestDAO.java:18]: they are inherited from the dao

File: src/main/java/p2/dao/IInterestDAO.java (hunk 2 above)
Comment (mjrerle) [reply src/main/java/p2/dao/IInterestDAO.java:22]: removed status from interest, why do we need status

File: src/main/java/p2/dao/impl/InterestDAO.java (hunk 4 above)
Comment (mjrerle) [reply src/main/java/p2/dao/impl/InterestDAO.java:23]: the table's name is Interests, the column name is product_id and the placeholder should be the same case as the column name

File: src/main/java/p2/dao/impl/UserDAO.java (hunk 5 above)
Comment (mjrerle) [reply src/main/java/p2/dao/impl/UserDAO.java:18]: the table name is Interests and I just removed the + for readability

**Review by Reiterpallasch — APPROVED** [review]
Looks to be good

**Pull request merged**